src/RequestInsurance/Controllers/RequestInsuranceEditController.php

```diff
@@ -0,0 +1,171 @@
+            'old_headers'          => $requestInsurance->getOriginal('headers'),
+            'new_headers'          => $requestInsurance->getOriginal('headers'),
+            'old_payload'          => $requestInsurance->getOriginal('payload'),
+            'new_payload'          => $requestInsurance->getOriginal('payload'),
+            'old_encrypted_fields' => $requestInsurance->encrypted_fields,
+            'new_encrypted_fields' => $requestInsurance->encrypted_fields,
+            'admin_user'           => $identityProvider->getUser($request),
+        ]);
+
+        return redirect()->back();
+    }
+
+    /**
+     * @param Request $request
+     * @param RequestInsuranceEdit $requestInsuranceEdit
+     * @param IdentityProvider $identityProvider
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function destroy(Request $request, RequestInsuranceEdit $requestInsuranceEdit, IdentityProvider $identityProvider)
+    {
+        // Only allow delete if not already applied and request is from the edit author
+        if ($requestInsuranceEdit->applied_at != null || $identityProvider->getUser($request) != $requestInsuranceEdit->admin_user) {
+            // Both these cases should not be possible from the view, so we don't send any error message
+            return redirect()->back();
+        }
+        $requestInsuranceEdit->delete();
+
+        return redirect()->back();
+    }
+
+    /**
+     * @param Request $request
+     * @param RequestInsuranceEdit $requestInsuranceEdit
+     * @param IdentityProvider $identityProvider
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function update(Request $request, RequestInsuranceEdit $requestInsuranceEdit, IdentityProvider $identityProvider)
+    {
+        // Only allow updates if it has not been applied and the updating user is the edit author
+        if ($requestInsuranceEdit->applied_at != null || $identityProvider->getUser($request) != $requestInsuranceEdit->admin_user) {
+            // Both these cases should not be possible from the view, so we don't send any error message
+            return redirect()->back();
+        }
+
+        DB::transaction(function () use ($request, $requestInsuranceEdit) {
+            // Remove all approvals
+            $requestInsuranceEdit->approvals()->delete();
+
+            // Update the edit
+            $requestInsuranceEdit->update([
+                'new_priority'         => $request->post('new_priority', $requestInsuranceEdit->new_priority),
+                'new_url'              => $request->post('new_url', $requestInsuranceEdit->new_url),
+                'new_method'           => $request->post('new_method', $requestInsuranceEdit->new_method),
+                'new_headers'          => $request->post('new_headers', ''),
+                'new_payload'          => $request->post('new_payload', ''),
+                'new_encrypted_fields' => $request->post('new_encrypted_fields', ''),
```

Comment:
Hvorfor bruger vi ikke fallback til de eksisterende værdier for de sidste 3 felter?

Reply:
Apropos encrypted fields... Har du tjekket at det er muligt at ændre i den, og at alt bliver encrypted og decrypted korrekt bagefter? :smile:

Reply:
Hvis jeg husker rigtigt er det fordi at hvis man sletter headers helt og sender, bliver det felt ikke inkluderet i post formen, så den vil i så fald bare beholde det gamle headers/payload i stedet for at slette det som ønsket.

Har ikke tjekket om det skaber problemer at edit encrypted fields.

Reply:
Som snakket om, de fields er unødvendige at edit, så har fjernet muligheden for at edit dem. Dog har jeg ikke lavet destructive migration der fjerner felterne helt.